### PR TITLE
Added internal SSL listener test

### PR DESCRIPTION
**Pull request by Kuvesz**
## Description

Added SSL support for internal listener E2E tests.

## Type of Change
- [x] Other (E2E testing)

## Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md)
- [x] I have verified this change is not present in other open pull requests
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

**Review by panyuenlau — DISMISSED** [review]
lgtm at first glance

File: tests/e2e/kafka.go [hunk 1]
```diff
@@ -48,3 +48,26 @@ func requireDeployingKafkaTopic(kubectlOptions k8s.KubectlOptions, topicName str
...
+// requireCreatingKafkaUser creates a KafkaUser resource from a template
+func requireCreatingKafkaUser(kubectlOptions k8s.KubectlOptions, userName string, tlsSecretName string) {
+	It("Deploying KafkaUser CR", func() {
+		templateParameters := map[string]interface{}{
+			"Name":      userName,
+			"Namespace": kubectlOptions.Namespace,
+		}
+		if tlsSecretName != "" {
+			templateParameters["TLSSecretName"] = tlsSecretName
+		}
```
Comment (pregnor) [inline tests/e2e/kafka.go:61]: **Note**: In the future we might want to extract these into a `kafkaUser(userName, namespace, optionalTLSSecretName string) (KafkaUser, error)` function, looks reusable.

File: tests/e2e/templates/user.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kafka.banzaicloud.io/v1alpha1
+kind: KafkaUser
+metadata:
+  name: {{or .Name "test-user"}}
+  namespace: {{or .Namespace "kafka" }}
+spec:
+  clusterRef:
+    name: kafka
```
Comment (pregnor) [inline tests/e2e/templates/user.yaml.tmpl:8]: **Question**: Shouldn't this be parametrized?

**Review by pregnor — APPROVED** [review]
LGTM at a glance.

File: tests/e2e/templates/user.yaml.tmpl (hunk 2 above)
Comment (Kuvesz) [reply tests/e2e/templates/user.yaml.tmpl:8]: oh yeah, I1ll do that real quick

File: tests/e2e/kafka.go (hunk 1 above)
Comment (Kuvesz) [reply tests/e2e/kafka.go]: Hm, fair, I'll add it to the internal refactor ticket if that is okay.

Comment (pregnor) [reply tests/e2e/kafka.go]: Sure

File: tests/e2e/templates/user.yaml.tmpl (hunk 2 above)
Comment (pregnor) [reply tests/e2e/templates/user.yaml.tmpl]: Thanks

**Review by pregnor — APPROVED** [review]

**Review by panyuenlau — APPROVED** [review]

**Review by pregnor — APPROVED** [review]

**Review by panyuenlau — APPROVED** [review]

**Pull request merged**